service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchPrecomputer.kt

```diff
@@ -11,17 +12,18 @@ import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import misk.hibernate.load
 import misk.logging.getLogger
+import okio.ByteString
 
 class BatchPrecomputer(
   private val backfillRunner: BackfillRunner
 ) {
+  private var pkeyCursor: ByteString? = backfillRunner.metadata.precomputingPkeyCursor
+  private var computedScannedRecordCount: Long = backfillRunner.metadata.computedScannedRecordCount
+  private var computedMatchingRecordCount: Long = backfillRunner.metadata.computedMatchingRecordCount
+
   fun run(coroutineScope: CoroutineScope) = coroutineScope.launch {
     logger.info { "BatchPrecomputer started ${backfillRunner.logLabel()}" }
 
-    // Start at the cursor we have in the DB, but after that we need to maintain our own,
```

Comment:
You don't like the comment?

Reply:
this comment was copied from `BatchQueuer` and is actually incorrect, because we always save the precomputing cursor, and we don't save the cursor in `BatchQueuer`, that one is saved after the RunBatch completes in `BatchAwaiter`

Reply:
So no? SGTM